build_web_compilers/lib/src/platforms.dart

```diff
@@ -37,13 +36,29 @@ const _additionalWebLibraries = [
 /// Additional libraries supported by dart2wasm.
 const _additionalWasmLibraries = ['ffi'];
 
-const _jsCompilerLibraries = [..._coreLibraries, ..._additionalWebLibraries];
-
-final ddcPlatform = DartPlatform.register('ddc', _jsCompilerLibraries);
-
-final dart2jsPlatform = DartPlatform.register('dart2js', _jsCompilerLibraries);
+// These intentionally throw if [initializePlatforms] wasn't called first.
+final ddcPlatform = DartPlatform.byName('ddc');
+final dart2jsPlatform = DartPlatform.byName('dart2js');
+final dart2wasmPlatform = DartPlatform.byName('dart2wasm');
 
-final dart2wasmPlatform = DartPlatform.register('dart2wasm', [
-  ..._coreLibraries,
-  ..._additionalWasmLibraries,
-]);
+/// Registers the platforms with [DartPlatform].
+///
+/// Must be called before [ddcPlatform], [dart2jsPlatform], or
+/// [dart2wasmPlatform] is used.
+void initializePlatforms([List<String> additionalCoreLibraries = const []]) {
```

Comment:
Is it important to pass in customizable allowed core libraries, or is a flag that enables a subset of them (like with `_additionalWebLibraries`) sufficient?

Reply:
I would only use it to set `ui` and `ui_web` and afaik there are no others, so yes we can also make this a bool flag.